src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/CompareConverter.shared.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public sealed class CompareConverter : CompareConverter<object>
+	{
+	}
+
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public abstract class CompareConverter<TObject> : ValueConverterExtension, IValueConverter
```

Comment:
@Cfun1 here, the `TObject` should implement `IComparable`?

Reply:
Yes, you are thinking to add the condition `where TControl : IComparable` ?

Reply:
Yes! You read my mind. I'll commit this change in a bit, I'm running here on my machine

Reply:
But if I want to do 

` public MyConverter : CompareConvertert<MyModel>`

I need to make sure that `MyModel : IComparable` no?

Reply:
So we should change the summary, because it says that object implements IComparable, or am I missing something?
"Converts an object that implements IComparable to an object or a boolean based on a comparaison."

Reply:
The object that needs to implement `IComparable` is alue and `ComparingValue`, while `TObject` is the type of `TrueObject` and `FalseObject` which are the returned objects, no comparison is made on those ones so no need for `IComparable`.

Reply:
I see now, thanks